Source/r_plane.c

```diff
@@ -423,6 +413,16 @@ static void do_draw_plane(visplane_t *pl)
         }
         else
         {
+          // Make sure the fade-to-color effect doesn't happen too early
+          if (dc_texturemid < SCREENHEIGHT / 2 * FRACUNIT)
+          {
+            fixed_t diff = dc_texturemid - SCREENHEIGHT / 2 * FRACUNIT;
+            diff %= textureheight[texture];
+            if (diff < 0)
```

Comment:
Will these lines need to get swapped? The division remainder is unlikely to be negative, or is it?

Reply:
`dc_texturemid` is often negative.

Reply:
Sure. It's just that I know this formula the other way round, i.e. first move to positive range then modulo. Like here: 
https://github.com/fabiangreffrath/woof/blob/758184e5bc9595c2b711f596bd38cc38629f69fe/Source/r_data.c#L511-L514